### PR TITLE
fix(cli): should proxy google social callback url properly to exp ui

### PR DESCRIPTION
**Pull request by charIeszhao**
## Summary

Fix the `isFileAsset()` util, in order to better support the Google social callback URL. Details can be found in attached Linear issue (LOG-9827).

## Testing

Locally tested. And Google social sign-in callback can be proxied properly

## Checklist

~~- [ ] `.changeset`~~

- [x] unit tests

~~- [ ] integration tests~~
~~- [ ] necessary TSDoc comments~~

### CODE REVIEW

File: packages/cli/src/commands/tunnel/utils.ts [hunk 1]
```diff
@@ -146,10 +146,14 @@ export const checkExperienceInput = (url?: string, staticPath?: string) => {
  * @example isLogtoRequestPath('/api/interaction/submit') // true
  * @example isLogtoRequestPath('/consent') // true
  */
-export const isLogtoRequestPath = (requestPath?: string) =>
+export const isLogtoRequestPath = (requestPath?: string): boolean =>
   ['/oidc/', '/api/'].some((path) => requestPath?.startsWith(path)) || requestPath === '/consent';
 
-const isFileAssetPath = (url: string) => url.split('/').at(-1)?.includes('.');
+const isFileAssetPath = (url: string): boolean => {
```
Comment (simeng-li) [inline packages/cli/src/commands/tunnel/utils.ts:152]: should have some unit tests?

Comment (charIeszhao) [reply packages/cli/src/commands/tunnel/utils.ts]: OK, added

File: packages/cli/src/commands/tunnel/utils.ts [hunk 2]
```diff
@@ -146,10 +146,14 @@ export const checkExperienceInput = (url?: string, staticPath?: string) => {
...
-export const isLogtoRequestPath = (requestPath?: string) =>
+export const isLogtoRequestPath = (requestPath?: string): boolean =>
   ['/oidc/', '/api/'].some((path) => requestPath?.startsWith(path)) || requestPath === '/consent';
 
-const isFileAssetPath = (url: string) => url.split('/').at(-1)?.includes('.');
+export const isFileAssetPath = (url: string): boolean => {
+  // Check if the request URL contains query params. If yes, ignore the params and check the request path
+  const pathWithoutQuery = url.split('?')[0];
+  return Boolean(pathWithoutQuery?.split('/').at(-1)?.includes('.'));
+};
```
Comment (gao-sun) [inline packages/cli/src/commands/tunnel/utils.ts:156]: `path.extname`?

Comment (charIeszhao) [reply packages/cli/src/commands/tunnel/utils.ts:156]: Tried, but it failed with the google social callback URL

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**